### ISSUE TITLE
createPost Issue: stuck at loading screen

### ISSUE THREAD

**Issue by Madgeniusblink** [dappuniversity/social-network#1]
Hi I was wondering why after creating the post the state sets to loading but never goes back to false(not loading)?

I try to debug it but no luck... I do get this error when I use the react chrome extension to check the state:

```
web3-core-method.umd.js:696 Uncaught TypeError: Duplicated method constructor. This method is defined as RPC call and as Object method.
    at Object.get (web3-core-method.umd.js:696)
    at s (backend.js:1)
    at c (backend.js:1)
    at c (backend.js:1)
    at c (backend.js:1)
    at h (backend.js:1)
    at Object.inspectElement (backend.js:6)
    at t.<anonymous> (backend.js:6)
    at t.r.emit (backend.js:6)
    at backend.js:30
    at t (backend.js:6)
```

Any idea what it could be? 
Here is my repo link: https://github.com/Madgeniusblink/SocialNetwork-Dapp

It seems that it also happens after user tips. 
Overall the application works just fine no issues there.. it just gets stuck in the loading state.. and I have no idea how to fix that.

Thank you in advance for taking a look at the issue!

**Comment by johnsin** [comment]
Yeah, I am running into the same problem.. It appears that there is something funny with the app.js.

`this.state.socialNetwork.methods.createPost(content).send({ from: this.state.account })
    .once('receipt', (receipt) => {
      console.log("Does This work");
     this.setState({ loading: false })
    })`

`this.state.socialNetwork.methods.tipPost(id).send({ from: this.state.account, value: tipAmount })
    .once('receipt', (receipt) => {
      this.setState({ loading: false })
    })`

It seems like the ".once" function isn't firing, or nothing is being returned.  I'm wondering if the web3.js updated something that blew this up.

**Comment by Madgeniusblink** [comment]
@johnsin  weird probably... I will try to work on it this week and see if I can get it to work the right way... Thank you for taking a look at the issue as well!

**Comment by johnsin** [comment]
@Madgeniusblink 
I'm starting here.. [https://web3js.readthedocs.io/en/v1.2.0/web3-eth-contract.html#once](https://web3js.readthedocs.io/en/v1.2.0/web3-eth-contract.html#once)..

But what is odd is that his code worked, and ours didn't.  So I am wondering if they updated the Web3 lib and when we pulled it down from out package.json.. we got a newer version that handles the .once/event/callback stuff differently.

I haven't tested anything yet... but this is what I am deducing the bug down to.  I'm just lazy right now.  ;)

If I get motivated.. ill update again!

**Comment by Madgeniusblink** [comment]
hahaha, thank you! and I will as well try it in next couple of days if you haven't gotten to it

**Comment by var-rohit** [comment]
@Madgeniusblink @johnsin  I have corrected the issue, most probably it was a version mismatch and have uploaded the project on https://github.com/var-rohit/Ethereum-DApp-using-React

**Comment by Legitao** [comment]
I updated web3 to the latest 1.2.6 and it worked
